### PR TITLE
Refactor/db layer

### PR DESCRIPTION
**Pull request by stevana**
Depends on PR #260.

### CODE REVIEW

**Comment by stevana** [comment]
Ready for review.

File: external-resolver/src/main/scala/org/genivi/sota/resolver/filters/FilterRepository.scala [hunk 1]
```diff
@@ -32,10 +33,13 @@ object FilterRepository {
   def add(filter: Filter)(implicit ec: ExecutionContext): DBIO[Filter] =
     (filters += filter).map(_ => filter)
 
-  def load(name: Filter.Name)(implicit ec: ExecutionContext): DBIO[Option[Filter]] =
+  def exists(name: Filter.Name)(implicit ec: ExecutionContext): DBIO[Filter] =
```
Comment (koshelev) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/filters/FilterRepository.scala]: I don't like failing DB action. For me it has semantic of failing db request, and it is not the case here

**Comment by stevana** [comment]
We've had this discussion many times; I've refactored the layer dealing with db/errors twice. I don't like it this way either, but I don't see any better way of doing it at the moment.

**Comment by koshelev** [comment]
Yes, and I believe it is the reason, why such errors were handled in *Functions classes

**Comment by stevana** [comment]
I can explain why that was a bad idea next time you are at work.

**Pull request merged**